### PR TITLE
Creating Post and Posts components

### PR DESCRIPTION
**Pull request by OmerSchechter**
#232

### CODE REVIEW

File: client/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -3,11 +3,16 @@ import { RouterModule, Routes } from '@angular/router';
...
+import { PostsComponent } from './components/posts/posts.component';
+import { PostComponent } from './components/post/post.component';
 
 const routes: Routes = [
   // { path: '', redirectTo: '/', pathMatch: 'full' }
   { path: '', component: PaiComponent },
   { path: 'about', component: AboutComponent },
+  { path: 'blog', component: PostsComponent },
+  { path: 'post/:id', component: PostComponent },
+
```
Comment (nirgn975) [inline client/src/app/app-routing.module.ts]: Remove new line

File: client/src/app/components/posts/posts.component.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	posts: Post[];
+
+  constructor(private postsService: PostsService) { }
+
+  ngOnInit(): void {
+  	this.getPosts();
+  }
+
+  getPosts(): void {
+  	this.posts = this.postsService.getPosts();
```
Comment (nirgn975) [inline client/src/app/components/posts/posts.component.ts]: Move this to `ngOnInit`. Remove the `getPosts` function

File: client/src/app/components/posts/posts.component.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import { Post } from '../../models/post';
+
+@Component({
+  selector: 'op-posts',
+  templateUrl: './posts.component.html',
+  styleUrls: ['./posts.component.scss'],
+  providers: [PostsService]
+})
+export class PostsComponent implements OnInit {
+
```
Comment (nirgn975) [inline client/src/app/components/posts/posts.component.ts]: Remove this whitespace

File: client/src/app/components/posts/posts.component.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	posts: Post[];
+
+  constructor(private postsService: PostsService) { }
```
Comment (nirgn975) [inline client/src/app/components/posts/posts.component.ts]: Move this service to a new line, like:
```ts
constructor(
  private postsService: PostsService
) { }
```

File: client/src/app/models/post.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export class Post {
+	unique_id: number;
```
Comment (nirgn975) [inline client/src/app/models/post.ts]: Fix indentation, use 2 spaces.

File: client/src/app/models/post.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+	title: string;
+	body: string;
+	author: string;
+	created_at: Date;
+	publish: Date;
+	tags: string[];
+}
```
Comment (nirgn975) [inline client/src/app/models/post.ts]: New line

File: client/src/app/services/posts.service.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import { Post } from '../models/post';
+import { Posts } from '../models/posts';
+
+@Injectable()
+export class PostsService {
+
+  constructor() { }
+
+  getPosts(): Post[] {
+
```
Comment (nirgn975) [inline client/src/app/services/posts.service.ts]: Remove this whitespace.

**Comment by OmerSchechter** [comment]
Made the requested changes.

File: client/src/app/app-routing.module.ts [hunk 8]
```diff
@@ -3,11 +3,15 @@ import { RouterModule, Routes } from '@angular/router';
...
 import { AboutComponent } from './components/about/about.component';
+import { PostsComponent } from './components/posts/posts.component';
+import { PostComponent } from './components/post/post.component';
 
 const routes: Routes = [
   // { path: '', redirectTo: '/', pathMatch: 'full' }
   { path: '', component: PaiComponent },
   { path: 'about', component: AboutComponent },
+  { path: 'blog', component: PostsComponent },
+  { path: 'post/:id', component: PostComponent },
```
Comment (nirgn975) [inline client/src/app/app-routing.module.ts]: Fix it `post/:id` to `blog/:id`

File: client/src/app/components/posts/posts.component.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  templateUrl: './posts.component.html',
+  styleUrls: ['./posts.component.scss'],
+  providers: [PostsService]
+})
+export class PostsComponent implements OnInit {
+	posts: Post[];
+
+  constructor(
+  	private postsService: PostsService
+	) { }
```
Comment (nirgn975) [inline client/src/app/components/posts/posts.component.ts]: Fix indentation

File: client/src/app/components/posts/posts.component.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  selector: 'op-posts',
+  templateUrl: './posts.component.html',
+  styleUrls: ['./posts.component.scss'],
+  providers: [PostsService]
+})
+export class PostsComponent implements OnInit {
+	posts: Post[];
+
+  constructor(
+  	private postsService: PostsService
```
Comment (nirgn975) [inline client/src/app/components/posts/posts.component.ts]: Fix indentation

File: client/src/app/components/posts/posts.component.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+	) { }
+
+  ngOnInit(): void {
+  	this.posts = this.postsService.getPosts();
```
Comment (nirgn975) [inline client/src/app/components/posts/posts.component.ts]: Fix indentation

**Pull request merged**